future-apps/payroll/contracts/Payroll.sol

```diff
@@ -435,55 +456,39 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
-     * @param _token Payment token
-     * @return Employee's payment allocation for token
+     * @notice Tells the payment proportion for an employee and a token.
+     * @param _employeeId Employee's identifier.
+     * @param _token Payment token to query the payment allocation of.
+     * @return Employee's payment allocation for the token being queried.
      */
-    function getAllocation(uint256 _employeeId, address _token) public view returns (uint8 allocation) {
+    function getAllocation(uint256 _employeeId, address _token) public view employeeIdExists(_employeeId) returns (uint256 allocation) {
         return employees[_employeeId].allocation[_token];
     }
 
     /**
-     * @dev Check if a token is allowed to be used in this app
-     * @param _token Token to check
-     * @return True if it's in the list of allowed tokens, false otherwise
+     * @dev Checks if a token is allowed to be used in this app.
+     * @param _token Address of the token to be checked.
+     * @return True if the given token is allowed, false otherwise.
      */
     function isTokenAllowed(address _token) public view returns (bool) {
         return allowedTokens[_token];
     }
 
-    /**
-     * @dev IForwarder interface conformance. Forwards any employee action.
-     * @param _evmScript script being executed
-     */
-    function forward(bytes _evmScript) public {
-        require(canForward(msg.sender, _evmScript), ERROR_NO_FORWARD);
-        bytes memory input = new bytes(0); // TODO: Consider input for this
-        address[] memory blacklist = new address[](1);
-        blacklist[0] = address(finance);
-        runScript(_evmScript, input, blacklist);
-    }
-
-    function canForward(address _sender, bytes) public view returns (bool) {
-        // Check employee exists (and matches)
-        return (employees[employeeIds[_sender]].accountAddress == _sender);
-    }
-
     // Internal fns
 
-    function _addEmployee(
-        address _accountAddress,
-        uint256 _initialDenominationSalary,
-        string _name,
-        string _role,
-        uint64 _startDate
-    )
-        internal
-    {
+    /**
+     * @notice Adds a new employee to Payroll.
+     * @param _accountAddress Employee's address to receive payroll.
+     * @param _initialDenominationSalary Employee's salary, per second in denomination token.
+     * @param _name Employee's name.
+     * @param _role Employee's role.
+     * @param _startDate Employee's starting timestamp in seconds.
+     */
+    function _addEmployee(address _accountAddress, uint256 _initialDenominationSalary, string _name, string _role, uint64 _startDate) internal {
         // Check address isn't already being used
-        require(employeeIds[_accountAddress] == 0, ERROR_EMPLOYEE_ALREADY_EXIST);
+        require(!_employeeExists(_accountAddress), ERROR_EMPLOYEE_ALREADY_EXIST);
 
+        // Employees start at index 1, to allow us to use employees[0] to check for non-existent address
```

Comment:
Oh, I see it here now, hehe. Anyway, I would keep it at the beginning, where it was, or in `initialize` function, where `nextEmployee` is set to 1.

Reply:
I think it makes much more sense to have it near to the LOC where we do it :)
I'll move it to the initialize function